multicluster/test/e2e/service_test.go

```diff
@@ -47,32 +47,51 @@ func testServiceExport(t *testing.T, data *TestData) {
 // we create a nginx in on cluster(east), and try to curl it in another cluster(west).
 // If we got status code 200, it means that the resources is exported by the east cluster
 // and imported by the west cluster.
-// TODO(yang): reorg test function contents
-func (data *TestData) testServiceExport(t *testing.T) {
-	podName := randName("test-nginx-")
-	clientPodName := "test-service-client"
 
-	if err := createPodWrapper(t, data, westCluster, multiClusterTestNamespace, podName, nginxImage, "nginx", nil, nil, nil, nil, false, nil); err != nil {
+func (data *TestData) setupTestResources(t *testing.T) {
+
+	if err := createPodWrapper(t, data, westCluster, multiClusterTestNamespace, nginxPodName, nginxImage, "nginx", nil, nil, nil, nil, false, nil); err != nil {
 		t.Fatalf("Error when creating nginx Pod in west cluster: %v", err)
 	}
-	defer deletePodWrapper(t, data, westCluster, multiClusterTestNamespace, podName)
 
-	if err := createPodWrapper(t, data, eastCluster, multiClusterTestNamespace, podName, nginxImage, "nginx", nil, nil, nil, nil, false, nil); err != nil {
+	if err := createPodWrapper(t, data, eastCluster, multiClusterTestNamespace, nginxPodName, nginxImage, "nginx", nil, nil, nil, nil, false, nil); err != nil {
 		t.Fatalf("Error when creating nginx Pod in east cluster: %v", err)
 	}
-	defer deletePodWrapper(t, data, eastCluster, multiClusterTestNamespace, podName)
 
 	if _, err := data.createService(westCluster, westClusterTestService, multiClusterTestNamespace, 80, 80, corev1.ProtocolTCP, map[string]string{"app": "nginx"}, false,
 		false, corev1.ServiceTypeClusterIP, nil, nil); err != nil {
 		t.Fatalf("Error when creating Servie %s in west cluster: %v", westClusterTestService, err)
 	}
-	defer deleteServiceWrapper(t, testData, westCluster, multiClusterTestNamespace, westClusterTestService)
 
 	if _, err := data.createService(eastCluster, eastClusterTestService, multiClusterTestNamespace, 80, 80, corev1.ProtocolTCP, map[string]string{"app": "nginx"}, false,
 		false, corev1.ServiceTypeClusterIP, nil, nil); err != nil {
 		t.Fatalf("Error when creating Servie %s in east cluster: %v", eastClusterTestService, err)
 	}
-	defer deleteServiceWrapper(t, testData, eastCluster, multiClusterTestNamespace, eastClusterTestService)
+}
+
+func (data *TestData) tearDownTestResources(t *testing.T) {
+	deleteServiceWrapper(t, testData, westCluster, multiClusterTestNamespace, westClusterTestService)
+	deleteServiceWrapper(t, testData, eastCluster, multiClusterTestNamespace, eastClusterTestService)
+	deletePodWrapper(t, data, westCluster, multiClusterTestNamespace, nginxPodName)
+	deletePodWrapper(t, data, eastCluster, multiClusterTestNamespace, nginxPodName)
+}
+
+func (data *TestData) getAndProbeFromCluster(t *testing.T, clusterName string){
```

Comment:
nit: need a better naming for this. Maybe `probeMCServiceFromCluster`?

Reply:
Yes. Let me rename it.